src/coprocessor/dag/expr/builtin_time.rs

```diff
@@ -462,6 +463,29 @@ impl ScalarFunc {
         let res = MyDuration::from_nanos(diff, d0.fsp().max(d1.fsp()) as i8)?;
         Ok(Some(Cow::Owned(res)))
     }
+
+    #[inline]
+    pub fn sub_duration_and_string<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &'a [Datum],
+    ) -> Result<Option<Cow<'a, MyDuration>>> {
+        let arg0: Cow<'a, MyDuration> = try_opt!(self.children[0].eval_duration(ctx, row));
```

Comment:
Actually... Since `MyDuration` is `Copy`, perhaps it's easier to just write 

```suggestion
        let arg0 = *try_opt!(self.children[0].eval_duration(ctx, row));
```

so those `arg3` and `into_owned()` can be eliminated?

Reply:
That's a lot better.